crates/subspace-farmer/src/dsn.rs

```diff
@@ -168,3 +175,34 @@ where
 
     Ok(())
 }
+
+/// Starts the relaying node and configure the client relay configuration.
+pub async fn configure_relay_server(listen_on: Vec<Multiaddr>) -> RelayConfiguration {
+    let internal_relay_node_address: Multiaddr =
+        Multiaddr::empty().with(Protocol::Memory(DEFAULT_RELAY_MEMORY_PORT));
+
+    let relay_settings = RelayLimitSettings::unlimited();
+
+    let config = Config {
+        listen_on,
+        allow_non_globals_in_dht: true,
+        relay_config: RelayConfiguration::Server(
+            internal_relay_node_address.clone(),
+            relay_settings,
+        ),
+        ..Config::with_generated_keypair()
+    };
+
+    let (node, mut node_runner) = subspace_networking::create(config)
+        .await
+        .expect("Relay node must be created");
+
+    trace!(node_id = %node.id(), "Relay Node started");
+
+    tokio::spawn(async move {
+        node_runner.run().await;
+    });
+
+    node.configure_relay_client()
+        .expect("We should have a valid relay client configuration here.")
```

Comment:
"Should" is a suggestion, it isn't guaranteed, it is not sufficient to justify panic here, so this should return an error.

Also it is probably a good idea to have a generic parameter in `Node` so we can distinguish nodes that have relay enabled at compile time (until we have a separate behavior for this).